src/views/List.jsx

```diff
@@ -1,15 +1,44 @@
+import { useState } from 'react';
+
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchTerm, setSearchTerm] = useState('');
+
+	const clearSearchTerm = () => {
+		setSearchTerm('');
+	};
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+			<form onSubmit={(e) => e.preventDefault()}>
+				<label>
+					Filter Items
+					<input
+						type="text"
+						placeholder="start typing here..."
+						id="filter"
+						name="filter"
+						value={searchTerm}
+						onChange={(e) => {
+							setSearchTerm(e.target.value);
+						}}
+					/>
+				</label>
+			</form>
+			<button onClick={clearSearchTerm}>clear</button>
+
 			<ul>
-				{data.map(({ name, id }) => (
-					<ListItem name={name} key={id} />
-				))}
+				{data
```

Comment:
Nice implementation! Doing it this way removes the need for a secondary filtered data `useState`.

One small consideration against putting the filter logic in the JSX might be that whenever the component rerenders for any reason it will have to run the filter function again.

Another easy suggestion for filter logic might be to drop in a [trim](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim) to remove whitespace from the ends of input strings.